lib/vectorsearch/pinecone.rb

```diff
@@ -25,12 +25,14 @@ def initialize(environment:, api_key:, index_name:, llm:, llm_api_key:)
 
     # Add a list of texts to the index
     # @param texts [Array] The list of texts to add
+    # @param namespace [String] The namespace to add the texts to
     # @return [Hash] The response from the server
-    def add_texts(texts:)
+    def add_texts(texts:, namespace: "")
       vectors = texts.map do |text|
         {
           # TODO: Allows passing in your own IDs
           id: SecureRandom.uuid,
+          namespace: namespace,
```

Comment:
Important note. The readme on the Pinecone client was out of date. The namespace needs to be outside the vectors array and passed as a named argument to the upsert method

```ruby
index = client.index(index_name)

index.upsert(
  namespace: namespace,
  vectors: vectors
)
```

Reply:
Nice catch, thanks! https://docs.pinecone.io/reference/upsert